patina_dxe_core/src/config_tables/memory_attributes_table.rs

```diff
@@ -7,30 +7,68 @@
 //! SPDX-License-Identifier: Apache-2.0
 //!
 extern crate alloc;
-use alloc::vec::Vec;
+use alloc::{boxed::Box, vec, vec::Vec};
+use zerocopy::{Immutable, IntoBytes};
 
 use core::{
     ffi::c_void,
     fmt::Debug,
     mem::size_of,
     slice,
-    sync::atomic::{AtomicBool, AtomicPtr, Ordering},
+    sync::atomic::{AtomicBool, Ordering},
 };
+use spin::Mutex;
 
 use crate::{
-    allocator::{MemoryDescriptorSlice, core_allocate_pool, core_free_pool, get_memory_map_descriptors},
+    allocator::{MemoryDescriptorSlice, get_memory_map_descriptors},
     config_tables::core_install_configuration_table,
     events::EVENT_DB,
     systemtables,
 };
 use r_efi::efi;
 
 // We cache the MAT here because we need to free it in whenever we get a new runtime code/data allocation
-static MEMORY_ATTRIBUTES_TABLE: AtomicPtr<c_void> = AtomicPtr::new(core::ptr::null_mut());
+static MEMORY_ATTRIBUTES_TABLE: Mutex<Option<MemoryAttributesTable>> = Mutex::new(None);
```

Comment:
This one we can use a TplMutex for. It probably isn't needed, but this does get triggered when we allocate runtime memory, which can happen during an interrupt.